docs: improve section on prCreation (#9571)
renovatebot/renovate: docs/usage/configuration-options.md

- `not-pending`: Renovate will wait until status checks have completed (passed or failed) before raising the PR
- `status-success`: Renovate won't raise PRs unless tests pass

Renovate defaults to `immediate` but you might want to change this to `not-pending` instead.

> With prCreation set to `immediate`, you'll get a GitHub notification right away when a new Renovate bot PR is avaiable.

Author: ```suggestion
With prCreation set to `immediate`, you'll get a GitHub notification right away when a new Renovate bot PR is available.
```

> With prCreation set to `immediate`, you'll get a GitHub notification right away when a new Renovate bot PR is available.

Reviewer: ```suggestion
With prCreation set to `immediate`, you'll get a Pull Request and possible associated notification right away when a new update is available.
```

Your test suite takes a bit of time to complete, so if you go look at the new PR right away, you don't know if your tests pass or fail.
You're basically waiting until you have the test results, before you can decide if you want to merge the PR or not.

> With prCreation set to `not-pending`, Renovate waits untill all test have finished running, and only then creates the PR.

Author: ```suggestion
With prCreation set to `not-pending`, Renovate waits until all test have finished running, and only then creates the PR.
```

> With prCreation set to `not-pending`, Renovate waits until all test have finished running, and only then creates the PR.

Author: ```suggestion
With prCreation set to `not-pending`, Renovate waits until all tests have finished running, and only then creates the PR.
```

Add missing `s`.
